ccswbs/gus
# Testimonial widget: null check and refactoring

# Summary of changes
Create a null check to resolve the following[ build error](https://app.netlify.com/sites/preview-ugconthub/deploys/65c102f368b910000808ceaf) that occurs when an empty testimonial widget is added to a page:

```
   error Truncated page data information for the failed page "/people/": {
   "errors": {},
   "path": "/people/",
   "slicesMap": {},
   "pageContext": {
     "id": "256770e6-15f6-57c3-9dce-a0d1cf43f977",
     "nid": "entity:node/2780",
     "tid": [
       "791cb49e-41a9-5a82-9bb1-7560894adcb3"
     ]
   }
 }
 failed Building static HTML for pages - 5.372s
 error Building static HTML failed for path "/people/"
 
    8 |   arrays.forEach((array) => {
    9 |     array.forEach((object) => {
 > 10 |       map[object[iteratee]] = object;
      |                 ^
   11 |     });
   12 |   });
```

## Frontend
- Add proper null check to `myUnionBy` function in `testimonial.js`
- Refactor and streamline testimonial code
- Replace tabs with spaces

[ x ] My changes are accessible (at minimum WCAG 2.0 Level AA)
[ x ] My changes are responsive and appear as expected on mobile and desktop views.
[ N/A ] After merging, I will update the [Content Hub documentation](https://uoguelphca.sharepoint.com/sites/UniversityContentHubInformationGroup) and ensure the Content Hub trainer understands how my changes will affect users.

## Backend
None (though we might consider some later on, e.g. why bother permitting multiple selections for Testimonial type and offering a "Staff" option when it's not being used)

# Test Plan

- Go to https://fixtestimonials--test-ugconthub.netlify.app/test-testimonials/ and verify the empty testimonial widgets do not render while the ones that have content do (backend version of this page is at https://tq-ginfix-chug.pantheonsite.io/node/2784/edit)
- Go to https://fixtestimonials--test-ugconthub.netlify.app/programs/bachelor-of-science-in-agriculture/ and verify all testimonial types display and are mentioned in the title 
- Go to https://fixtestimonials--test-ugconthub.netlify.app/programs/bachelor-of-engineering/ and verify the single Student testimonial displays as expected

## Known Issues

If you go to https://fixtestimonials--test-ugconthub.netlify.app/programs/bachelor-of-bio-resource-management/ the testimonial title is incorrect - it only mentions alumni when it should also mention students. However, this bug occurs on production as well and is unrelated to this PR: https://www.uoguelph.ca/programs/bachelor-of-bio-resource-management/
